comp/logs/agent/config/processing_rules.go

```diff
@@ -64,6 +67,9 @@ func ValidateProcessingRules(rules []*ProcessingRule) error {
 // CompileProcessingRules compiles all processing rule regular expressions.
 func CompileProcessingRules(rules []*ProcessingRule) error {
 	for _, rule := range rules {
+		if rule.Type == ExcludeTruncated {
+			continue
+		}
```

Comment:
Suggestion for the future: we could consider decoupling the deserialization from rule initialization so that `ProcessingRule` becomes an `interface` where it can be responsible for its own "compilation" to avoid this increasing complex conditional flow based on type. That is `ProcessingRule` would turn into something like:

```
type ProcessingRule interface {
  Compile() error
  Validate() error
  Match(*message.Message) (bool, error)
}
```

There would need to be a different deserialization struct to read the config into that is the union of all of the possible fields.

Again, no need to change this now, just a suggestion for future refactoring :)